tesseract_command_language/include/tesseract_command_language/cartesian_waypoint.h

```diff
@@ -169,8 +171,18 @@ class CartesianWaypoint
   Eigen::VectorXd upper_tolerance;
 
   /**
-   * @brief Returns true if waypoint is toleranced
-   * @return True if waypoint is toleranced
+   * @brief Allow storing a seed associated with this cartesian waypoint
+   * @details Using the waypoint type erasure type to allow user defined types.
+   * The most common type that would be used it the joint waypoint for the following
+   *   - providing a seed to the IK solver
+   *   - providing a seed to the IK solver with modified limits using the tolerances
+   *   - providing a state to be used in a simple planner for interpolation to avoid running IK
+   */
+  Waypoint seed{ NullWaypoint() };
```

Comment:
Is there a particular use-case where the seed wouldn't be a joint waypoint? I'm wondering if it would be beneficial to be more restrictive here for clarity

Reply:
That is what I originally had but thought it would be nice to use the type erasure type so a user defined joint waypoint could be used.